LaraApi/src-lara/lara/vitishls/VitisHls.js

```diff
@@ -0,0 +1,153 @@
+        this.vitisDir = "VitisHLS";
+        this.vitisProjName = "VitisHLSClavaProject";
+    }
+
+    setTopFunction(topFunction) {
+        this.topFunction = topFunction;
+
+        return this;
+    }
+
+    setPlatform(platform) {
+        this.platform = platform;
+        return this;
+    }
+
+    setClock(clock) {
+        if (clock <= 0) {
+            throw new Error("[" + this.toolName + "] Clock value must be a positive integer!");
+        } else {
+            this.clock = clock;
+        }
+        return this;
+    }
+
+    addSource(file) {
+        this.sourceFiles.push(file);
+    }
+
+    synthesize(verbose = true) {
+        println("[" + this.toolName + "] Setting up Vitis HLS executor");
+
+        this.clean();
+        this.#generateTclFile();
+        this.#executeVitis(verbose);
+        return Io.isFile(this.#getSynthesisReportPath());
+    }
+
+    clean() {
+        Io.deleteFolderContents(this.vitisDir);
+    }
+
+    #getSynthesisReportPath() {
+        var path = this.vitisDir + "/" + this.vitisProjName;
+        path += "/solution1/syn/report/csynth.xml";
+        return path;
+    }
+
+    #executeVitis(verbose) {
+        println("[" + this.toolName + "] Executing Vitis HLS");
+
+        const pe = new ProcessExecutor();
+        pe.setWorkingDir(this.vitisDir);
+        pe.setPrintToConsole(verbose);
+        pe.execute("vitis_hls", "-f", "script.tcl");
+
+        println("[" + this.toolName + "] Finished executing Vitis HLS");
+    }
+
+    #getTclInputFiles() {
+        var str = "";
+        const weavingFolder = ToolUtils.parsePath(Clava.getWeavingFolder());
+
+        // make sure the files are woven
+        Io.deleteFolderContents(weavingFolder);
+        Clava.writeCode(weavingFolder);
+
+        // if no files were added, we assume that every woven file should be used
+        if (this.sourceFiles.length == 0) {
+            println("[" + this.toolName + "] No source files specified, assuming current AST is the input");
+            for (var file of Io.getFiles(Clava.getWeavingFolder())) {
+                const exts = [".c", ".cpp", ".h", ".hpp"];
+                const res = exts.some(ext => file.name.includes(ext));
+                if (res)
+                    str += "add_files " + weavingFolder + "/" + file.name + "\n"
+            }
+        }
+        else {
+            for (const file of this.sourceFiles) {
+                str += "add_files " + weavingFolder + "/" + file + "\n";
+            }
+        }
+        return str;
+    }
+
+    #generateTclFile() {
+        var cmd = "open_project " + this.vitisProjName + "\n";
+        cmd += "set_top " + this.topFunction + "\n";
+        cmd += this.#getTclInputFiles();
+        cmd += "open_solution \"solution1\" -flow_target vitis\n";
+        cmd += "set_part {" + this.platform + "}\n";
+        cmd += "create_clock -period " + this.clock + " -name default\n";
+        cmd += "csynth_design\n";
+        cmd += "exit\n";
```

Comment:
I think this code could be improved by using JS Template Literals instead ( https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Template_literals )

Reply:
Done